xdsl/transforms/riscv_scf_loop_fusion.py

```diff
@@ -0,0 +1,87 @@
+from typing import cast
+
+from xdsl.dialects import builtin, riscv, riscv_scf
+from xdsl.ir import MLContext
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    op_type_rewrite_pattern,
+)
+from xdsl.transforms.canonicalization_patterns.riscv import get_constant_value
+
+
+class FuseNestedLoopsPattern(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: riscv_scf.ForOp, rewriter: PatternRewriter) -> None:
+        if op.iter_args:
+            return
+
+        outer_body = op.body.block
+        if not isinstance(inner_loop := outer_body.first_op, riscv_scf.ForOp):
+            # Outer loop must contain inner loop
+            return
+        if inner_loop is not cast(riscv_scf.YieldOp, outer_body.last_op).prev_op:
+            # Outer loop must contain only inner loop and yield
+            return
+        if inner_loop.iter_args:
+            return
+
+        if (inner_lb := get_constant_value(inner_loop.lb)) is None:
+            return
+        if inner_lb.value.data != 0:
+            return
+
+        if (inner_ub := get_constant_value(inner_loop.ub)) is None:
+            return
+        if (outer_step := get_constant_value(op.step)) is None:
+            return
```

Comment:
Why only fuse on constant bounds/step? Can't you emit instructions that calculate the new bounds?